docs/build/tools/foundry.mdx

```diff
@@ -0,0 +1,246 @@
+on-chain data and precompiles (rng, signatures, on-chain encryption, etc.).
+
+Follow the steps below to setup your project:
+
+1. Install and run Foundryup:
+
+   ```shell
+   curl -L https://foundry.paradigm.xyz | bash
+   foundryup
+   source ~/.bashrc
+   ```
+
+2. Create a new Forge project and move inside:
+
+   ```shell
+   forge init sapphire_demo
+   cd sapphire_demo
+   ```
+
+3. After intializing the project, we can install the **Sapphire Contracts**
+package using Foundry package manager [soldeer]:
+
+   ```shell
+   forge soldeer install @oasisprotocol-sapphire-contracts~0.2.11
+   ```
+
+### Installing Sapphire Foundry (Optional)
+
+If you intend to use Sapphire features in Forge tests,
+follow these additional steps:
+
+1. Install the **Sapphire Foundry** soldeer package:
+
+   ```shell
+   forge soldeer install @oasisprotocol-sapphire-foundry~0.1.0
+   ```
+
+2. Add `ffi = true` to your foundry.toml file:
+
+3. Your foundry.toml file should now look like this:
+
+   ```toml
+   [profile.default]
+   src = "src"
+   out = "out"
+   libs = ["lib", "dependencies"]
+   ffi = true
+
+   [dependencies]
+   "@oasisprotocol-sapphire-contracts" = "0.2.11"
+   "@oasisprotocol-sapphire-foundry" = "0.1.0"
+   ```
+
+   Note:
+   - `ffi = true` enables `vm.ffi()` ([foreign function interface])
+   which is used to call rust bindings containing the precompile and
+   decryption logic
+   - `"dependencies"` lists the required project dependencies installed
+   via the Foundry package manager [soldeer]
+   - (Alternative) You can also skip previous steps, copy the contents
+   to your foundry.toml file and run:
+
+   ```shell
+   forge soldeer install
+   ```
+
+4. Install rust toolchain (nightly):
+
+   ```shell
+   curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+   rustup toolchain install nightly
+   rustup default nightly
+   ```
+
+   Note: `@oasisprotocol-sapphire-foundry` relies on Oasis Sapphire
+   `runtime-sdk` crate which requires nightly toolchain.
+
+5. Install Sapphire precompiles:
+   ```shell
+   cd dependencies/@oasisprotocol-sapphire-foundry-0.1.0/precompiles
+   cargo +nightly build --release
+   ```
+
+## Sapphire specific features
+
+Users who aren't familiar with confidential benefits of Sapphire
+should take a moment to review the Ethereum [comparison page].
+The `@oasisprotocol-sapphire-foundry` package enables
+testing with precompiles and encrypted calls.
+
+#### Precompiles
+
+Sapphire precompiles are special precompiled contracts that
+contain cryptographic primitives and enable confidential transactions.
+Since Foundry uses un-customized revm, they are not available by default.
+
+To enable them in Forge tests, add the **import** statement to your test file:
+
+```solidity
+import {SapphireTest} from "@oasisprotocol-sapphire-foundry-0.1.0/BaseSapphireTest.sol";
+```
+and then inherit from `SapphireTest` from `BaseSapphireTest.sol`
+and override the `setUp()` function:
+```solidity
```

Comment:
````suggestion
```

and then inherit from `SapphireTest` from `BaseSapphireTest.sol`
and override the `setUp()` function:

```solidity
````

Reply:
Updated docs